File: Data Structures & Algorithms/remove-element/submission-1.py
class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        
        l=0

        for r in range(len(nums)):
            if nums[r] != val:
                nums[l]=nums[r]
                l+=1
        return l
        # i=0

        # while i < len(nums):

        #     if nums[i]==val:
        #         nums.pop(i)
        #     else:
        #         i+=1
        # return len(nums)
        # for i in range(len(nums)-1,-1,-1):
        #     if nums[i]==val:
        #         nums.pop(i)
        # return len(nums)

File: Data Structures & Algorithms/top-k-elements-in-list/submission-1.rs
impl Solution {
    pub fn top_k_frequent(nums: Vec<i32>, k: i32) -> Vec<i32> {

        let mut maxheap = BinaryHeap::new();

        let mut hashmap = HashMap::new();


        for i in 0..nums.len(){
            *hashmap.entry(nums[i]).or_insert(0) += 1;
        }
        
        for (key,val) in hashmap {
            maxheap.push((val,key));
        }

        let mut result = vec![];
        for _ in 0..k{
            if let Some((val,key)) = maxheap.pop(){
            result.push(key)
        }
        }

        result

            
        






    }
}
